### DISCUSSION TITLE
Arithmetic operations and SQL functions applied to DATE and DATETIME types

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#793] [Ideas]
To fully benefit from the available DATE and DATETIME types, I think it would be useful to be able to:

- perform arithmetic operations on DATE and DATETIME types
- apply SQL functions to DATE and DATETIME types

Currently, it is not possible to perform arithmetic operations, for example subtracting two DATETIME values (the result is always null).
It is also not possible to apply SQL functions, for example `min()` (the result is always null).
This is true for both the old (`java.util.Date`) and new (`java.time.LocalDateTime`) implementations.

Background: I'm migrating a project from Hsqldb to ArcadeDB and I have spent a lot of time trying to migrate all SQL statements from the Hsqldb sql dialect to the ArcadeDB sql dialect.
In some cases the migration was easy, in other cases I had to add some post-processing code (for example to migrate a Self Join), but there is one last SQL statement which I simply cannot find a way to migrate:
```
SELECT 
  file_name, 
  start_validity, 
  stop_validity 
FROM 
  (
    SELECT 
      file_name, 
      file_type, 
      start_validity, 
      stop_validity, 
      version, 
      ABS(TIMESTAMPDIFF(SQL_TSI_SECOND, TIMESTAMP (stop_validity), TIMESTAMP (CAST(? AS TIMESTAMP)))) as dist 
    FROM 
      product 
    where 
      file_type = ?
  ) 
ORDER BY dist LIMIT 1
```
The purpose of this query is to find the row that has the closest stop_validity to the Timestamp provided as positional parameter to the prepared statement.

**Comment by lvca** [comment]
@vic0824 100% agree with this. OrientDB allows to do some math with dates, so I assumed we have test cases from this inherited from the OrientDB project. This should be pretty straightforward to do. Creating an issue from this discussion: #794
